MREUnityRuntime/MREUnityRuntimeLib/Assets/AssetSource.cs

```diff
@@ -5,26 +5,29 @@
 namespace MixedRealityExtension.Assets
 {
     /// <summary>
-    /// Documents the origin of an asset
+    /// Documents the origin of an asset.
+    /// NOTE: The InternalId is not used for equality testing.
     /// </summary>
-    public struct AssetSource
+    public class AssetSource
     {
         /// <summary>
         /// The type of container the asset came from.
         /// </summary>
-        public AssetContainerType ContainerType { get; }
+        public AssetContainerType ContainerType { get; set; }
 
         /// <summary>
         /// The URL of the asset's container.
         /// </summary>
-        public Uri Uri { get; }
+        public Uri Uri { get; set; }
```

Comment:
This needs to be a string. If the URL is malformed it will fail the JSON serialization (Uri class will fail to parse it). That's too early to be performing field validation.